securedrop/source_app/__init__.py

```diff
@@ -139,57 +132,17 @@ def load_instance_config() -> None:
     @app.before_request
     @ignore_static
     def setup_g() -> Optional[werkzeug.Response]:
-        """Store commonly used values in Flask's special g object"""
-
         if 'expires' in session and datetime.utcnow() >= session['expires']:
-            msg = render_template('session_timeout.html')
-
-            # Show expiration message only if the user was
-            # either in the codename generation flow or logged in
-            show_expiration_message = any([
-                session.get('show_expiration_message'),
-                logged_in(),
-                was_in_generate_flow(),
-            ])
-
-            # clear the session after we render the message so it's localized
-            session.clear()
-
-            # Persist this properety across sessions to distinguish users whose sessions expired
-            # from users who never logged in or generated a codename
-            session['show_expiration_message'] = show_expiration_message
-
-            # Redirect to index with flashed message
-            if session['show_expiration_message']:
-                flash(Markup(msg), "important")
-            return redirect(url_for('main.index'))
+            # If the session as has expired, redirect to index with flashed message
+            # if the user was in the middle of the generate flow
+            if 'codenames' in session:
+                return clear_session_and_redirect_to_logged_out_page(session)
 
         session['expires'] = datetime.utcnow() + \
             timedelta(minutes=getattr(config,
                                       'SESSION_EXPIRATION_MINUTES',
                                       120))
 
-        # ignore_static here because `crypto_util.hash_codename` is scrypt
-        # (very time consuming), and we don't need to waste time running if
-        # we're just serving a static resource that won't need to access
-        # these common values.
```

Comment:
This is no longer needed since I removed the "time consuming" scrypt call from the code that handles incoming requests.